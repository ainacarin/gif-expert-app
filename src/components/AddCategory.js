import React, { useState } from "react";
import PropTypes from 'prop-types';
import "./AddCategory.css";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const [error, setError] = useState(null);


  const resetError = () => setError('');

  const handleChangeInput = (event) => {
    resetError();
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    // validation no empty input
    if (inputValue.trim().length > 1) {
      setCategories((oldCategories) => [...oldCategories, inputValue]);
      // clean input
      setInputValue("");
    } else{
      setError('Error value');
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={inputValue}
          onChange={handleChangeInput}
          placeholder="Write new character"
        ></input>
        {error && <p>{error}</p>}
      </form>
    </>
  );
};

// Proptypes definition
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}