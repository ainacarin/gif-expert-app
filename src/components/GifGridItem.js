import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, title, url } ) => {

    return (
        <div id={id} className='card animate__animated animate__fadeIn'>
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    );

};

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}