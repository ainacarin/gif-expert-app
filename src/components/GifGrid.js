import React, { useState, useEffect } from "react";
import { getGifs } from "../api/giphyApi";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

    // exect only when dependencies are modified
    useEffect(() => {
      getGifs( category ).then(setImages);
    }, [category]);
    //   // exect only once time, the first time component
    //   useEffect(() => {
    //     getGifs( category ).then( setImages );
    //   }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
