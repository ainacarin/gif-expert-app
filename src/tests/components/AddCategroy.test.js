import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('AddCategory Group', () => {

    // ref fn to evaluate (times calls, etc)
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    });

    test('AddCategory: setCategory', () => {

        expect( wrapper ).toMatchSnapshot();

    });


    /** User actions */
    test('AddCategory: input onChange correct', () => {

        const input = wrapper.find('input');
        const value = 'Hello World';
        input.simulate('change', { target: { value } });

        const resInput = wrapper.find('input');
        expect(resInput.prop('value')).toBe(value);
    });

    test('AddCategory: no character input -> <p>error</p>', () => {

        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault(){} });
        const errorElem = wrapper.find('p');
        const errorText = 'Error value';
        const input = form.find('input');
        expect(input.prop('value')).toBe('');
        expect(errorElem.text()).toBe(errorText);
        expect( setCategories ).not.toHaveBeenCalled( );
    });


});