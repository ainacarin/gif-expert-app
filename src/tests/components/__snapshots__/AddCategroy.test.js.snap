// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Group AddCategory: setCategory 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Write new character"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
